File: src/main/java/gargoyle/xenox/util/log/LEVEL.java
package gargoyle.xenox.util.log;

public enum LEVEL {
    FATAL("FATAL", 0),
    ERROR("ERROR", 1),
    INFO("INFO", 2),
    WARN("WARN", 3),
    DEBUG("DEBUG", 4);
    private final int level;
    private final String string;

    LEVEL(String string, int level) {
        this.string = string;
        this.level = level;
    }

    public boolean canLog(LEVEL level) {
        return this.level >= level.level;
    }

    @Override
    public String toString() {
        return string == null ? "" : string;
    }
}

File: src/main/kotlin/gargoyle/xenox/util/log/LEVEL.kt
package gargoyle.xenox.util.log

enum class LEVEL(string: String, level: Int) {
    FATAL("FATAL", 0), ERROR("ERROR", 1), INFO("INFO", 2), WARN("WARN", 3), DEBUG("DEBUG", 4);

    private val level: Int
    private val string: String?

    init {
        this.string = string
        this.level = level
    }

    fun canLog(level: LEVEL?): Boolean {
        return this.level >= level!!.level
    }

    override fun toString(): String {
        return string ?: ""
    }
}
